org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleBuildSupport.java

```diff
@@ -58,10 +64,16 @@ public void update(IProject project, boolean force, IProgressMonitor monitor) th
 		}
 		JavaLanguageServerPlugin.logInfo("Starting Gradle update for " + project.getName());
 		if (force) {
-			String projectPath = project.getLocation().toFile().getAbsolutePath();
-			BuildConfiguration buildConfiguration = GradleProjectImporter.getBuildConfiguration(Paths.get(projectPath));
-			GradleBuild build = GradleCore.getWorkspace().createBuild(buildConfiguration);
-			build.synchronize(monitor);
+			Optional<GradleBuild> build = GradleCore.getWorkspace().getBuild(project);
+			if (build.isPresent()) {
+				GradleBuild gradleBuild = build.get();
+				if (isRoot(project, gradleBuild, monitor)) {
+					String projectPath = project.getLocation().toFile().getAbsolutePath();
+					BuildConfiguration buildConfiguration = GradleProjectImporter.getBuildConfiguration(Paths.get(projectPath));
+					gradleBuild = GradleCore.getWorkspace().createBuild(buildConfiguration);
+				}
+				gradleBuild.synchronize(monitor);
+			}
 		} else {
```

Comment:
this `else` condition branch is redundant, and it's logical is similar as the `if (force) { }` branch. Please consider merging the duplicated logic. 

For example:
```java
Optional<GradleBuild> build = GradleCore.getWorkspace().getBuild(project);
if (build.isPresent()) {
  GradleBuild gradleBuild = build.get();
 if (force && isRoot(project, gradleBuild, monitor)) {
  ...
  }

gradleBuild.synchronize(monitor);
}
```

Reply:
Fixed.